### PR TITLE
add pnnl data parser

### LINKED ISSUE #563: Add PNNL site and 2017 and 2018 data to reference database
Will be used for used for Topic Area 2 evaluations. We don't need to establish an automatically updating data fetch.

### PR DESCRIPTION
**Pull request by wholmgren**
- [x] Closes #563  .
  - [x] I am familiar with the [contributing guidelines](https://solarforecastarbiter-core.readthedocs.io/en/latest/contributing.html).
  - [x] Tests added.
  - [x] Updates entries to [`docs/source/api.rst`](https://github.com/SolarArbiter/solarforecastarbiter-core/blob/master/docs/source/api.rst) for API changes.
  - [x] Adds descriptions to appropriate "what's new" file in [`docs/source/whatsnew`](https://github.com/SolarArbiter/solarforecastarbiter-core/tree/master/docs/source/whatsnew) for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` or this Pull Request with `` :pull:`num` ``. Includes contributor name and/or GitHub username (link with `` :ghuser:`user` ``).
  - [x] New code is fully documented. Includes [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant docstrings, examples, and comments where necessary.
  - [x] Maintainer: Appropriate GitHub Labels and Milestone are assigned to the Pull Request and linked Issue.

### CODE REVIEW

**Comment by wholmgren** [comment]
Works but 1 year of data exceeds the API limit.

```
2021-04-16 14:13:49,472 INFO Updating PNNL ghi from 2017-01-01 00:00:00+00:00 to 2018-01-01 00:00:00+00:00.
2021-04-16 14:13:49,611 DEBUG Posting data to PNNL ghi between 2016-12-31 16:00:00-08:00 and 2017-12-31 15:59:00-08:00.
2021-04-16 14:13:53,701 DEBUG Resetting dropped connection: dev-api.solarforecastarbiter.org
2021-04-16 14:13:56,087 DEBUG https://dev-api.solarforecastarbiter.org:443 "POST /observations/8abae936-9ef7-11eb-881b-0a580a8003fa/values HTTP/1.1" 413 80
2021-04-16 14:13:56,087 ERROR Posting data to PNNL ghi failed.
2021-04-16 14:13:56,087 DEBUG HTTP Error: {"errors":{"413":"Payload Too Large. Maximum payload size is 16777216 bytes."}}
```

**Comment by wholmgren** [comment]
3 months works

```
solararbiter referencedata update --network 'PNNL' --base-url 'https://dev-api.solarforecastarbiter.org' --start 20170101 --end 2017040
```

File: solarforecastarbiter/io/reference_observations/pnnl.py [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+from pathlib import Path
+
+import pandas as pd
+
+from solarforecastarbiter.io.reference_observations import (
+    common, default_forecasts)
+
+logger = logging.getLogger('reference_data')
+
+DATA_PATH = Path('pnnl_data')
```
Comment (alorenzo175) [inline solarforecastarbiter/io/reference_observations/pnnl.py:13]: pretty restrictive, maybe make it come from an environment variable?

File: solarforecastarbiter/io/reference_observations/pnnl.py [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+    """
+    common.create_forecasts(api, site, VARIABLE_MAP.values(),
+                            default_forecasts.TEMPLATE_FORECASTS)
+
+
+def fetch(api, site, start, end):
+    """Retrieve observation data for PNNL site for 1 year.
+
+    Assumes that data is available in a directory structure
+    ``pnnl_data/rldradC1.00.{%Y}/rldradC1.00.%Y%m%d.%H%M%S.raw.%Y%m%D%Hpnl.sky``
```
Comment (alorenzo175) [inline solarforecastarbiter/io/reference_observations/pnnl.py:91]:
```suggestion
    ``DATA_PATH/rldradC1.00.{%Y}/rldradC1.00.%Y%m%d.%H%M%S.raw.%Y%m%D%Hpnl.sky``
```

File: solarforecastarbiter/io/reference_observations/pnnl.py [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+    ----------
+    api : io.APISession
+        Unused but conforms to common.update_site_observations call
+    site : datamodel.Site
+        Unused but conforms to common.update_site_observations call
+    start : datetime
+        The beginning of the period to request data for.
+    end : datetime
+        The end of the period to request data for. If ``end.year >
+        start.year``, data will only be pulled through ``start.year``.
```
Comment (alorenzo175) [inline solarforecastarbiter/io/reference_observations/pnnl.py:103]: Why? seems not so hard to loop through multiple years. I understand one might not be able to post that much at once

File: solarforecastarbiter/io/reference_observations/pnnl.py (hunk 1 above)
Comment (wholmgren) [reply solarforecastarbiter/io/reference_observations/pnnl.py:13]: will you use an environment variable when you import data?

Comment (alorenzo175) [reply solarforecastarbiter/io/reference_observations/pnnl.py:13]: yes. can still default to the current value

File: solarforecastarbiter/cli.py [hunk 4]
```diff
@@ -52,8 +52,13 @@ def cli_access_token(user, password):
...
     except requests.HTTPError as e:
-        click.echo(click.style(
-            e.response.json()['error_description'], fg='red'))
+        try:
+            message = e.response.json()['error_description']
+        except json.decoder.JSONDecodeError:
+            message = e.response.text
+            if not message and e.response.status_code == 500:
+                message = 'Internal Server Error'
+        click.echo(click.style(message, fg='red'))
```
Comment (wholmgren) [inline solarforecastarbiter/cli.py:61]: When trying to understand the unhandled JSONDecodeError exception, I accidentally committed a print statement here. I figured I might as well clean up the exception when removing the print statement, but didn't put much thought into it. I can revert if that's better.

```solararbiter referencedata update --network 'PNNL' --base-url 'https://dev-api.solarforecastarbiter.org' --user "testing@solarforecastarbiter.org" --password=Thepassword123! --start 20171231 --end 20180102 -vvv
2021-04-20 12:52:37,364 DEBUG Starting new HTTPS connection (1): solarforecastarbiter.auth0.com:443
2021-04-20 12:52:37,604 DEBUG https://solarforecastarbiter.auth0.com:443 "POST /oauth/token HTTP/1.1" 500 0
Internal Server Error
```

**Review by alorenzo175 — APPROVED** [review]

File: solarforecastarbiter/io/tests/test_api.py [hunk 5]
```diff
@@ -1256,6 +1261,7 @@ def auth_token():
                   '"password": "Thepassword123!", '
                   '"audience": "https://api.solarforecastarbiter.org", '
                   '"client_id": "c16EJo48lbTCQEhqSztGGlmxxxmZ4zX7"}'))
+        token_req.raise_for_status()
```
Comment (wholmgren) [inline solarforecastarbiter/io/tests/test_api.py:1264]: @alorenzo175 more scope creep but this fixes the tests when the api/auth0 isn't available
